src/core/mac/mac.cpp

```diff
@@ -1349,7 +1581,12 @@ void Mac::RecordFrameTransmitStatus(const TxFrame &aFrame,
 
 void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError)
 {
-    if (!aFrame.IsEmpty())
+#if OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
```

Comment:
Thanks. Yes, we can. Will change it in next push.

For longer term, I like to actually move the broadcast retx logic to sub-MAC layer (since we already do all other retx in sub-MAC or radio). This way it would be directly tied to 15.4 link.